### DISCUSSION TITLE
Release 2.4

### DISCUSSION THREAD

**Discussion by revoxhere** [revoxhere/duino-coin#387] [General]
### Hello everyone! This update addresses a lot of the bug fixes and improvements that have been done over the past two weeks.

Changelog:
* PC Miner:
  * Fixed reconnecting :spider_web: 
  * Improved colored output :rainbow: 
  * Improved debug output :bug: 
  * Improved error handling :heavy_check_mark: 
  * Improved DUCO-S1 speed :racehorse: 
  * Some code cleanups and optimizations :broom:
  * Added new xxhash algorithm :new:
  * Added new translations :world_map: 
  * Other small fixes and improvements

* AVR Miner & Code - **please upload the new Arduino Code to your board!**:
  * New communication system :new:
  * Fixed reconnecting :spider_web: 
  * Fixed hashrate reporting :bar_chart: 
  * Improved colored output :rainbow: 
  * Improved debug output :bug:
  * Improved error handling :heavy_check_mark: 
  * Improved DUCO-S1A speed (up to 10% higher hashrates) :racehorse: 
  * Some code cleanups and optimizations :broom:
  * Other small fixes and improvements

* CLI & GUI Wallets:
  * Binded the enter key to some basic actions
  * Removed the buggy system tray support
  * Small improvements and some bugfixes :sagittarius: 

* ESP8266/ESP32 Codes:
  * Small improvements :sagittarius: 

### ⚠️ NOTICE: .exe binaries may be detected as a virus by your antivirus software. This is a **false positive** caused by pyinstaller bootloader, read about it [here](https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg). Duino-Coin is not a virus.

For the next release we're planning to ship the pool system. You can now also use the new [Duino-Coin Android app](https://play.google.com/store/apps/details?id=com.pripun.duinocoin) made by Pripun. For now I'm going to take my time rewriting the server to make it more stable. See you in the next update and happy mining :wave:

<hr /><em>This discussion was created from the release <a href='https://github.com/revoxhere/duino-coin/releases/tag/2.4'>Release 2.4</a>.</em>

**Comment by prasittanaprakob** [comment]
I tried the ESP8266 code, but it said the cannot find the Crypto.h. Then I got Crypto.h from the internet then another error "error: 'experimental' has not been declared"

how to solve it?

**Comment by ghost** [comment]
I'm using release 2.4 and when I'm starting AVR_Miner.py on my MacBook Pro it gives me this ArributeError. I can see it has something to do with the new language that came in version 2.3, but unfortunately I don't know exactly how to fix it. Can somebody take a look at this please so I can mine from my MacBook :-) 

Daniels-MacBook-Pro:duino-coin-2.4 danieljovar$ python3 AVR_Miner.py
Traceback (most recent call last):
  File "/Users/danieljovar/Downloads/duino-coin-2.4/AVR_Miner.py", line 132, in <module>
    if locale.startswith("es"):
AttributeError: 'NoneType' object has no attribute 'startswith'

Thanks in advance!

**Reply by ghost** [reply]
And sudo doesn't fix it!

**Reply by revoxhere** [reply]
Oh, looks like macOS handkes locales weirdly, I will push a fix soon

**Comment by bitro-de** [comment]
Hey, can you make a version to mine on a MacBook Pro too? Right now i use webminer, but i would prefer a miner that always mines in background when using my laptop. Thank you!

**Reply by revoxhere** [reply]
I don't have macOS to compile the binaries for it. If you want to, you can launch the programs from the source code.

**Comment by prasittanaprakob** [comment]
hello,

I use esp8266 for mining with miner version 2.4.6 and I get 4.56 khash/s. Is the speed ok?

**Reply by eb1991** [reply]
When you upload the code make sure that you set clock to 160 (default is 80) I am mining with esp8266 at 9.2 kH/s

**Reply by prasittanaprakob** [reply]
thank you.

**Comment by prasittanaprakob** [comment]
is Duino server up and running 24/7 because during the day like in the morning 7.00-14.00 everything not responds.

the serial monitor of esp8266 say: Bad not enough data 1.
